core/time/timezone/tzdate.odin

```diff
@@ -0,0 +1,145 @@
+package timezone
+
+import "core:fmt"
+import "core:time"
+import "core:time/datetime"
+
+region_load :: proc(reg: string, allocator := context.allocator) ->  (out_reg: ^datetime.TZ_Region, ok: bool) {
+	return _region_load(reg, allocator)
+}
+
+region_destroy :: proc(region: ^datetime.TZ_Region) {
+	if region == nil {
+		return
+	}
+
+	for name in region.shortnames {
+		delete(name)
+	}
+	delete(region.shortnames)
+	delete(region.records)
+	delete(region.name)
+	free(region)
+}
+
+region_get_nearest :: proc(region: ^datetime.TZ_Region, tm: time.Time) -> datetime.TZ_Record {
+	n := len(region.records)
+	left, right := 0, n
+
+	tm_sec := time.to_unix_seconds(tm)
+	for left < right {
+		mid := int(uint(left+right) >> 1)
+		if region.records[mid].time < tm_sec {
+			left = mid + 1
+		} else {
+			right = mid
+		}
+	}
+
+	idx := max(0, left-1)
+	return region.records[idx]
+}
+
+datetime_to_utc :: proc(dt: datetime.DateTime) -> (out: datetime.DateTime, success: bool) #optional_ok {
+	if dt.tz == nil {
+		return dt, true
+	}
+
+	tm := time.datetime_to_time(dt) or_return
+	record := region_get_nearest(dt.tz, tm)
+
+	secs := time.time_to_unix(tm)
+	adj_time := time.unix(secs - record.utc_offset, 0)
+	adj_dt := time.time_to_datetime(adj_time) or_return
+	return adj_dt, true
+}
+
+datetime_to_tz :: proc(dt: datetime.DateTime, tz: ^datetime.TZ_Region) -> (out: datetime.DateTime, success: bool) #optional_ok {
+	dt := dt
+	if dt.tz == tz {
+		return dt, true
+	}
+
+	if dt.tz != nil {
+		dt = datetime_to_utc(dt)
+	}
+
+	if tz == nil {
+		return datetime_to_utc(dt)
+	}
+
+	tm := time.datetime_to_time(dt) or_return
+	record := region_get_nearest(tz, tm)
+
+	secs := time.time_to_unix(tm)
+	adj_time := time.unix(secs + record.utc_offset, 0)
+	adj_dt := time.time_to_datetime(adj_time) or_return
+	adj_dt.tz = tz
+
+	return adj_dt, true
+}
+
+
+shortname :: proc(dt: datetime.DateTime) -> (name: string, success: bool) #optional_ok {
+	tm, ok := time.datetime_to_time(dt)
+	if !ok { return }
+
+	if dt.tz == nil { return "UTC", true }
+
+	record := region_get_nearest(dt.tz, tm)
+	return record.shortname, true
+}
+
+shortname_unsafe :: proc(dt: datetime.DateTime) -> string {
```

Comment:
May want to add some docs to say what the difference is to the safe version and when it's okay to use.